Update: Remove blank lines at beginning of files (fixes #5045)

jaredsohn opened the pull request:
Update no-multiple-empty-lines rule to also warn about blank lines at the beginning of a file. (For issue #5045)

platinumazure commented:
Since this is an enhancement of an existing rule, I think your commit message should be prefixed with "Update:".

nzakas commented:
Yup, @jaredsohn can you please update your commit summary to say "Update:" at the beginning?

Also, please add some tests to verify your changes.

jaredsohn commented:
I have changed the commit to say "Update:" and added tests. 

I tried to model the BOF tests after the EOF ones. The EOF template string tests didn't seem to actually be testing template strings or anything new so I removed them. (Template strings start and end with a nonwhitespace character so I don't think those tests make sense.)

nzakas commented:
Please add those tests back. Template strings can span multiple lines and empty lines in template strings shouldn't be flagged. 

Also, please update your commit message to say "fixes" instead of "refs".

jaredsohn commented:
>Please add those tests back. Template strings can span multiple lines and empty lines in template strings shouldn't be flagged.

In the latest commit, I am modifying the old template string tests for maxEOF to actually use template strings and creating similar maxBOF tests. (I don't have a lot of familiarity with them, but it seems like they need to start/end with the '`' symbol and the old tests did not.) I think the template string tests are far more important for checking the max parameter than these others but am adding the additional tests for completeness sake.

I have also updated the commit message to say 'fixes'.

nzakas commented:
Lgtm but I'd like @alberto to take a look (I think he made the last changes to this)

alberto commented:
Sorry, I can't until the end of next week. At first sight, it's missing the docs for the new option.

jaredsohn commented:
I am working on the documentation and hope to update it soon. I just want to note that I think I found an error in documentation unrelated to my changes. The first example says '/*error Multiple blank lines not allowed.*/' but max is set to 2; I will be changing this example to set max to 1.

jaredsohn commented:
I have updated the documentation. A couple of things I noticed:

The first example says '/*error Multiple blank lines not allowed.*/' but max is set to 2; I changed this example to set max to 1
If I am interpreting things correctly, the '10 extra lines' comment should be moved up a line, so I have done that.

nzakas commented:
Lgtm. Thanks
